src/test/java/uk/gov/pay/api/utils/ConnectorMockClient.java

```diff
@@ -60,11 +63,14 @@ private String createChargeResponse(long amount, String chargeId, String status,
                 .add("return_url", returnUrl)
                 .add("payment_provider", paymentProvider)
                 .add("created_date", createdDate)
-                .add("links", asList(links))
-                .build();
+                .add("links", asList(links));
+        if (gatewayTransactionId != null) {
+            jsonStringBuilder.add("gateway_transaction_id", gatewayTransactionId);
+        }
```

Comment:
I guess this is because we are letting jackson map the object to the json (with the IgnoreUnknownProperties). Fair enough having some test case around it as well.